pennylane/ops/qubit/hamiltonian.py

```diff
@@ -371,12 +371,7 @@ def simplify(self):
             c = self.coeffs[i]
             op = op if isinstance(op, Tensor) else Tensor(op)
 
-            ind = None
-            for j, o in enumerate(new_ops):
-                if op.compare(o):
-                    ind = j
-                    break
```

Comment:
🥲

Reply:
[Sourcery](https://sourcery.ai/) is amazing... 🙄

Reply:
I'll have to check this out for myself